plugins/gcloud/lib/samson_gcloud/image_tagger.rb

```diff
@@ -38,6 +40,16 @@ def tag_image(tag, base, digest, job_output)
           #{output.strip}
         TEXT
         job_output.puts "FAILED" unless success
+        success
+      end
+
+      def cache_last_tagged(key, value)
+        old = Rails.cache.read(key)
+        if old == value
+          Rails.cache.write(key, value) # refresh the cache expiration
```

Comment:
actually a race condition here but doubt that will happen very often ... and if it does it will be fixed with the next version